tests/ui/mir/alignment/addrof_alignment.rs

```diff
@@ -1,5 +1,4 @@
 // run-pass
-// ignore-wasm32-bare: No panic messages
 // compile-flags: -C debug-assertions
 
 struct Misalignment {
```

Comment:
Would be cleaner to use `byte_add` instead of `cast.add.cast`, IMO.

Also -- I am confused; doesn't this test show that `addr_of!` alignment was already ignored before this PR? What's the behavior change?